classes/indoorFloor.tsx

```diff
@@ -39,59 +45,174 @@ class IndoorFloor {
-                color = 'blue';
+        if (type === "toilet") {
+            return {
+                zIndex: 7,
+                position: 'absolute',
+                top: yPosition,
+                left: xPosition,
+                borderRadius: 25 / 2,
+                width: 25,
+                height: 25,
+                opacity: 1,
             }
-        })
-        if(this.floorData.entrance && xIndex === this.floorData.entrance.x && yIndex === this.floorData.entrance.y) {
-            color = 'white'
-        }
-        if(this.floorData.elevator && xIndex === this.floorData.elevator.x && yIndex === this.floorData.elevator.y) {
-            color = 'black'
+
         }
-        if(this.floorData.bathrooms) {
-            this.floorData.bathrooms.forEach(bathroom =>{
-                if(xIndex === bathroom.x && yIndex === bathroom.y) {
-                    color = 'yellow'
-                }
-            })
+        else if (type === "waterfountain"){
+            return {
+                zIndex: 7,
+                position: 'absolute',
+                top: yPosition,
+                left: xPosition,
+                borderRadius: 23 / 2,
+                width: 23,
+                height: 23,
+                opacity: 1
+            }
         }
-        if(this.floorData.waterFountains) {
-            this.floorData.waterFountains.forEach(waterFountain =>{
-                if(xIndex === waterFountain.x && yIndex === waterFountain.y) {
-                    color = 'purple'
-                }
-            })
+        else if (type === "elevator"){
+            return {
+                zIndex: 7,
+                    position: 'absolute',
+                    top: yPosition,
+                    left: xPosition,
+                    width: 25,
+                    height: 25,
+                    opacity: 1
+            }
         }
-        if(this.floorData.classRooms) {
-            this.floorData.classRooms.forEach(classRoom =>{
-                if(xIndex === classRoom.location.x && yIndex === classRoom.location.y) {
-                    color = 'orange'
-                }
-            })
+        else if (type === "staircase"){
+            return {
+                zIndex: 7,
+                    position: 'absolute',
+                    top: yPosition,
+                    left: xPosition,
+                    width: 25,
+                    height: 25,
+                    opacity: 1
+            }
         }
-        return color;
+        return {}
     }
 
+    // this function is called when a set of coordina  i - z return a match with the TYPE of indoor poit
+    // this function returns an image (logo) of the specified indoor poi
+    // function is called in showIndoorTile()
+    // this function calls setLogoStyle to apply the correct style to each type of icon
+    setIndoorPOILogo(i, z, xPosition, yPosition, type) {
```

Comment:
I would call this 'setIndoorPOIImage' or 'setInfoorPOIIcon' as  Logo isn't what we are displaying and can be confusing to find if looking for this segment of code